src/core/main.c

```diff
@@ -147,9 +149,52 @@ static NUMAPolicy arg_numa_policy;
 /* A copy of the original environment block */
 static char **saved_env = NULL;
 
+static volatile sig_atomic_t flag_self_healing_reexec = 0;
+static sigjmp_buf self_healing_jmp_buf;
+
 static int parse_configuration(const struct rlimit *saved_rlimit_nofile,
                                const struct rlimit *saved_rlimit_memlock);
 
+static void reexec_handler(int sig) {
+        flag_self_healing_reexec = 1;
+}
+
+static int do_self_healing(void) {
+        struct sigaction sa;
+        sigset_t ss;
+        int r = 0;
+
+        sigemptyset(&ss);
+        sigaddset(&ss, SIGTERM);
+        sigprocmask(SIG_UNBLOCK, &ss, NULL);
+
+        sa.sa_handler = reexec_handler;
+        sa.sa_flags = SA_RESTART;
+        sigaction(SIGTERM, &sa, NULL);
+
+        log_emergency("CrashSelfHealing state entered.");
+        log_emergency("Please analyze the collected coredump and upgrade systemd to the bug fixed version.");
```

Comment:
Remove "please" from the messages. Nice is nice, but brevity in emergency messages is more important ;)
Maybe "Fatal crash, PID1 halted. Send SIGTERM to PID1 to trigger reexecution."
And then explain about the need to collect the coredump and upgrade systemd in the man page.